doc/src/modules/polys/numberfields.rst

```diff
@@ -0,0 +1,289 @@
+8. the structure of the class group $Cl(K)$
+9. decide whether a given ideal is principal, and if so compute a generator.
+
+As a foundation, and to support our basic ability to define and work with
+number fields and algebraic numbers, we also set the following problems:
+
+10. Given an algebraic number -- expressed by radicals and rational operations,
+    or even as a special value of a transcendental function -- determine its
+    minimal polynomial over $\mathbb{Q}$.
+11. Given an irreducible polynomial, distinguish its roots as elements of
+    $\mathbb{C}$ by computing isolating intervals.
+12. The Primitive Element Problem: Given several algebraic numbers
+    $\alpha_1, \ldots, \alpha_m$, compute a single algebraic number $\theta$
+    such that $\mathbb{Q}(\alpha_1, \ldots, \alpha_m) = \mathbb{Q}(\theta)$.
+13. The Field Isomorphism Problem: Decide whether two number fields
+    $\mathbb{Q}(\alpha)$, $\mathbb{Q}(\beta)$ are isomorphic.
+14. The Field Membership Problem: Given two algebraic numbers $\alpha$,
+    $\beta$, decide whether $\alpha \in \mathbb{Q}(\beta)$, and if so write
+    $\alpha = f(\beta)$ for some $f(x) \in \mathbb{Q}[x]$.
+
+At present only a subset of the tasks enumerated in the lists above is yet
+supported in SymPy, and if you are interested in expanding support, you are
+encouraged to contribute!
+
+At time of writing, the existing solutions to the above problems are found
+in the following places:
+
+=================================  ======================================
+Task                               Implementation
+=================================  ======================================
+(1) integral basis                 :py:func:`~.round_two`
+(2) prime decomposition            :py:func:`~.prime_decomp`
+(3) $\mathfrak{p}$-adic valuation  :py:func:`~.prime_valuation`
+(10) find min poly                 :py:func:`~.minpoly`
+(11) isolating intervals           :py:func:`~.isolate`
+(12) primitive element             :py:func:`~.primitive_element`
+(13) field isomorphism             :py:func:`~.field_isomorphism`
+(14) field membership              :py:func:`~.to_number_field`
+=================================  ======================================
+
+
+Internals
+=========
+
+Algebraic number fields
+-----------------------
+
+Algebraic number fields are represented in SymPy by the
+:py:class:`~.AlgebraicField` class, which is a part of the
+polynomial domains system.
+
+
+
+Representing algebraic numbers
+------------------------------
+
+There are several different ways to represent algebraic numbers, and different
+forms may be preferable for different computational tasks.
+See [Cohen93]_ Section 4.2.
+
+
+As number field elements
+````````````````````````
+
+In SymPy, there is a distinction between number and expression classes defined
+in the :py:mod:`sympy.core.numbers` module on the one hand, and domains and
+domain elements defined in the :py:mod:`~sympy.polys` module on the other.
```

Comment:
This is explained in more detail in
```
:ref:`polys-domainsintro`
```
which links to here:
https://docs.sympy.org/latest/modules/polys/domainsintro.html